src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +42,38 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid array is passed,
+                // so we need to do extra logic to identify this scenario using JArray.Parse(input).
+
+                // If input contains '[' or ']', we would like to analyze it further as these markers for a Json data array.
+                if ((input.IndexOf("[", StringComparison.OrdinalIgnoreCase) >= 0) ||
+                    (input.IndexOf("]", StringComparison.OrdinalIgnoreCase) >= 0))
+                {
+                    // Remove carriage return, new-line and empty space from the string.
+                    string stringToValidate = input.Replace(System.Environment.NewLine, string.Empty);
+                    stringToValidate = stringToValidate.Replace(" ", string.Empty);
```

Comment:
Do we care about other white spaces? For example what if there is `\t`?
By the way, new lines are white spaces too.